src/core/string_map.cc

```diff
@@ -101,6 +101,74 @@ sds StringMap::Find(std::string_view key) {
   return GetValue(str);
 }
 
+void StringMap::RandomPair(sds* key, sds* value) {
+  auto itr = begin();
+  for (unsigned int i = 0; i < (rand() % Size()); ++i)
+    ++itr;
+  *key = itr->first;
+  if (value != nullptr)
+    *value = itr->second;
+  return;
+}
+
+unsigned int StringMap::RandomPairsUnique(unsigned int count, sds* keys, sds* vals) {
```

Comment:
the function might have other use cases in the future... so I will use your other suggestion to make it uint32_t

Reply:
There will be no other use cases for this function as the API is pretty stable. If it does change then we will change it accordingly. Otherwise, each time I have to look on that code I need to wonder why the return value is ignored. 

Also maybe just for our sanity, you can add a `DCHECK` that the size of the result vectors are the size of count (although I checked your algorithm -- just to be on the safe side -- extra checks don't do harm and are a safery barrier)

Reply:
> Also maybe just for our sanity, you can add a `DCHECK` that the size of the result vectors are the size of count (although I checked your algorithm -- just to be on the safe side -- extra checks don't do harm and are a safery barrier)

Currently in RandomPairs, I have the following code

```
  keys.reserve(count);
  if (with_value)
    vals.reserve(count);
```

This sort of enforced the size constraint. 

But I will add the DCHECK in RandomPairsUnique.

Reply:
> There will be no other use cases for this function as the API is pretty stable. If it does change then we will change it accordingly. Otherwise, each time I have to look on that code I need to wonder why the return value is ignored.

changed.